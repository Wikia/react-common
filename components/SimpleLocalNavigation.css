/**
* Implemenation taken from http://www.antimath.info/css/sass-sqrt-function/
*/
/**
* Implementation taken from http://alienryderflex.com/hsp.html
*/
/**
* Implemenation taken from http://www.antimath.info/css/sass-sqrt-function/
*/
/**
* Implementation taken from http://alienryderflex.com/hsp.html
*/
/**
* Implemenation taken from http://www.antimath.info/css/sass-sqrt-function/
*/
/**
* Implementation taken from http://alienryderflex.com/hsp.html
*/
/**
* Implemenation taken from http://www.antimath.info/css/sass-sqrt-function/
*/
/**
* Implementation taken from http://alienryderflex.com/hsp.html
*/
/**
* Implemenation taken from http://www.antimath.info/css/sass-sqrt-function/
*/
/**
* Implementation taken from http://alienryderflex.com/hsp.html
*/
.wds-fandom-content-well {
  box-sizing: border-box;
}
@media only screen and (max-width: 767px) {
  .wds-fandom-content-well {
    margin: 0 18px;
  }
}
@media only screen and (min-width: 768px) {
  .wds-fandom-content-well {
    margin: 0 auto;
  }
}
@media only screen and (min-width: 768px) and (max-width: 1023px) {
  .wds-fandom-content-well {
    width: 732px;
  }
}
@media only screen and (min-width: 1024px) and (max-width: 1279px) {
  .wds-fandom-content-well {
    width: 988px;
  }
}
@media only screen and (min-width: 1280px) {
  .wds-fandom-content-well {
    width: 1244px;
  }
}

.simple-local-navigation {
  background: #fff;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.18);
}
.simple-local-navigation__item {
  align-items: center;
  border-bottom: 5px solid transparent;
  color: #002a32;
  cursor: pointer;
  display: inline-flex;
  font-size: 14px;
  font-weight: 500;
  height: 50px;
  justify-content: center;
  letter-spacing: 0.2px;
  padding-top: 5px;
  text-decoration: none;
  text-transform: uppercase;
}
.simple-local-navigation__item:not(:last-child) {
  margin-right: 24px;
}
.simple-local-navigation__item:visited, .simple-local-navigation__item:active {
  color: inherit;
}
.simple-local-navigation__item:hover {
  color: #005252;
}
.simple-local-navigation__item.is-active {
  border-color: #088488;
  color: #088488;
}