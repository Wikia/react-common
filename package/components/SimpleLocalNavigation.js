'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react'));
var PropTypes = _interopDefault(require('prop-types'));

function _classCallCheck(instance, Constructor) {
  if (!(instance instanceof Constructor)) {
    throw new TypeError("Cannot call a class as a function");
  }
}

function _defineProperties(target, props) {
  for (var i = 0; i < props.length; i++) {
    var descriptor = props[i];
    descriptor.enumerable = descriptor.enumerable || false;
    descriptor.configurable = true;
    if ("value" in descriptor) descriptor.writable = true;
    Object.defineProperty(target, descriptor.key, descriptor);
  }
}

function _createClass(Constructor, protoProps, staticProps) {
  if (protoProps) _defineProperties(Constructor.prototype, protoProps);
  if (staticProps) _defineProperties(Constructor, staticProps);
  return Constructor;
}

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

function _inherits(subClass, superClass) {
  if (typeof superClass !== "function" && superClass !== null) {
    throw new TypeError("Super expression must either be null or a function");
  }

  subClass.prototype = Object.create(superClass && superClass.prototype, {
    constructor: {
      value: subClass,
      writable: true,
      configurable: true
    }
  });
  if (superClass) _setPrototypeOf(subClass, superClass);
}

function _getPrototypeOf(o) {
  _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) {
    return o.__proto__ || Object.getPrototypeOf(o);
  };
  return _getPrototypeOf(o);
}

function _setPrototypeOf(o, p) {
  _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) {
    o.__proto__ = p;
    return o;
  };

  return _setPrototypeOf(o, p);
}

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

function _objectWithoutProperties(source, excluded) {
  if (source == null) return {};

  var target = _objectWithoutPropertiesLoose(source, excluded);

  var key, i;

  if (Object.getOwnPropertySymbols) {
    var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

    for (i = 0; i < sourceSymbolKeys.length; i++) {
      key = sourceSymbolKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
      target[key] = source[key];
    }
  }

  return target;
}

function _assertThisInitialized(self) {
  if (self === void 0) {
    throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
  }

  return self;
}

function _possibleConstructorReturn(self, call) {
  if (call && (typeof call === "object" || typeof call === "function")) {
    return call;
  }

  return _assertThisInitialized(self);
}

function createCommonjsModule(fn, module) {
	return module = { exports: {} }, fn(module, module.exports), module.exports;
}

var classnames = createCommonjsModule(function (module) {
/*!
  Copyright (c) 2017 Jed Watson.
  Licensed under the MIT License (MIT), see
  http://jedwatson.github.io/classnames
*/
/* global define */

(function () {

	var hasOwn = {}.hasOwnProperty;

	function classNames () {
		var classes = [];

		for (var i = 0; i < arguments.length; i++) {
			var arg = arguments[i];
			if (!arg) continue;

			var argType = typeof arg;

			if (argType === 'string' || argType === 'number') {
				classes.push(arg);
			} else if (Array.isArray(arg) && arg.length) {
				var inner = classNames.apply(null, arg);
				if (inner) {
					classes.push(inner);
				}
			} else if (argType === 'object') {
				for (var key in arg) {
					if (hasOwn.call(arg, key) && arg[key]) {
						classes.push(key);
					}
				}
			}
		}

		return classes.join(' ');
	}

	if (module.exports) {
		classNames.default = classNames;
		module.exports = classNames;
	} else {
		window.classNames = classNames;
	}
}());
});

/**
 * FandomContentWell wraps `children` in `wds-content-well($use-xxlarge-breakpoint: false)` CSS mixin.
 */

var FandomContentWell = function FandomContentWell(_ref) {
  var className = _ref.className,
      children = _ref.children,
      rest = _objectWithoutProperties(_ref, ["className", "children"]);

  return React.createElement("div", _extends({
    className: "wds-fandom-content-well ".concat(className)
  }, rest), children);
};

FandomContentWell.propTypes = {
  /** @ignore */
  children: PropTypes.node,

  /** Additional class name */
  className: PropTypes.string
};
FandomContentWell.defaultProps = {
  children: null,
  className: ''
};

/**
 * SimpleLocalNavigation is a simple navigation than can be used with links or `react-router`.
 *
 * To mark child as active use exported static `SimpleLocalNavigation.ACTIVE_CLASS_NAME`.
 */

var SimpleLocalNavigation =
/*#__PURE__*/
function (_React$PureComponent) {
  _inherits(SimpleLocalNavigation, _React$PureComponent);

  function SimpleLocalNavigation() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, SimpleLocalNavigation);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(SimpleLocalNavigation)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "addClassName", function (child) {
      var className = classnames(child.props.className, 'simple-local-navigation__item');
      var props = {
        className: className
      };
      return React.cloneElement(child, props);
    });

    return _this;
  }

  _createClass(SimpleLocalNavigation, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var className = classnames('simple-local-navigation', this.props.className);
      return React.createElement("div", {
        className: className
      }, React.createElement(FandomContentWell, null, React.Children.map(this.props.children, function (child) {
        return _this2.addClassName(child);
      })));
    }
  }]);

  return SimpleLocalNavigation;
}(React.PureComponent);

_defineProperty(SimpleLocalNavigation, "ACTIVE_CLASS_NAME", 'is-active');

SimpleLocalNavigation.propTypes = {
  /** ignore */
  children: PropTypes.node.isRequired,

  /** Additional class name */
  className: PropTypes.string
};
SimpleLocalNavigation.defaultProps = {
  className: undefined
};

module.exports = SimpleLocalNavigation;
