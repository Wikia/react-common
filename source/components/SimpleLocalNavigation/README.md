Open this example in the new page to see the difference:

```js
const { default: SimpleLocalNavigation, ACTIVE_CLASS_NAME } = require('./index');

class SimpleLocalNavigationTest1 extends React.Component {
    constructor() {
        this.state = {
            value: 0,
        };
        this.onChange = this.onChange.bind(this);
    }

    onChange(value) {
        this.setState({value});
    }

    render() {
        return (
            <SimpleLocalNavigation>
                <span
                    className={this.state.value === 0 ? ACTIVE_CLASS_NAME : ''}
                    onClick={() => this.onChange(0)}
                >
                    Foo
                </span>
                <span
                    className={this.state.value === 1 ? ACTIVE_CLASS_NAME : ''}
                    onClick={() => this.onChange(1)}
                >
                    Bar
                </span>
                <span
                    className={this.state.value === 2 ? ACTIVE_CLASS_NAME : ''}
                    onClick={() => this.onChange(2)}
                >
                    Third option
                </span>
            </SimpleLocalNavigation>
        );
    }
}

<SimpleLocalNavigationTest1 />
```
