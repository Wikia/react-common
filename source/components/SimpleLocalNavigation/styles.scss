@import "~@fandom/design-system/dist/scss/wds-variables/index";
@import "../FandomContentWell/styles.scss";

.simple-local-navigation {
	background: #fff;
	box-shadow: 0 2px 4px 0 rgba(0, 0, 0, .18);

	&__item {
		align-items: center;
		border-bottom: 5px solid transparent;
		color: $wds-fandom-color-navy;
		cursor: pointer;
		display: inline-flex;
		font-size: 14px;
		font-weight: $wds-font-weight-medium;
		height: 50px;
		justify-content: center;
		letter-spacing: .2px;
		padding-top: 5px;
		text-decoration: none;
		text-transform: uppercase;

		&:not(:last-child) {
			margin-right: 24px;
		}

		&:visited,
		&:active {
			color: inherit;
		}

		&:hover {
			color: $wds-fandom-color-link-hover;
		}

		&.is-active {
			border-color: #088488;
			color: #088488;
		}
	}
}
