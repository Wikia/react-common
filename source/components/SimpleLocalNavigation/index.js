import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

// eslint-disable-next-line no-restricted-imports
import FandomContentWell from '../FandomContentWell';

import './styles.scss';

/**
 * SimpleLocalNavigation is a simple navigation than can be used with links or `react-router`.
 *
 * To mark child as active use exported static `SimpleLocalNavigation.ACTIVE_CLASS_NAME`.
 */
class SimpleLocalNavigation extends React.PureComponent {
    static ACTIVE_CLASS_NAME = 'is-active';

    addClassName = (child) => {
        const className = classNames(
            child.props.className,
            'simple-local-navigation__item',
        );

        const props = {
            className,
        };

        return React.cloneElement(child, props);
    }

    render() {
        const className = classNames(
            'simple-local-navigation',
            this.props.className,
        );

        return (
            <div className={className}>
                <FandomContentWell>
                    {React.Children.map(this.props.children, child => this.addClassName(child))}
                </FandomContentWell>
            </div>
        );
    }
}


SimpleLocalNavigation.propTypes = {
    /** ignore */
    children: PropTypes.node.isRequired,
    /** Additional class name */
    className: PropTypes.string,
};

SimpleLocalNavigation.defaultProps = {
    className: undefined,
};


export default SimpleLocalNavigation;
