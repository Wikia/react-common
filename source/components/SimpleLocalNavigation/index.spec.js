import { mount } from 'enzyme';
import React from 'react';

import SimpleLocalNavigation from './index';

test('SimpleLocalNavigation renders', () => {
    const component = mount(
        <SimpleLocalNavigation className="foo">
            <span>Item 1</span>
            <span>Item 2</span>
        </SimpleLocalNavigation>
    );
    expect(component).toMatchSnapshot();
});
