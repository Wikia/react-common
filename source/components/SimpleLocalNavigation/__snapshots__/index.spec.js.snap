// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleLocalNavigation renders 1`] = `
<SimpleLocalNavigation
  className="foo"
>
  <div
    className="simple-local-navigation foo"
  >
    <FandomContentWell
      className=""
    >
      <div
        className="wds-fandom-content-well "
      >
        <span
          className="simple-local-navigation__item"
          key=".0"
        >
          Item 1
        </span>
        <span
          className="simple-local-navigation__item"
          key=".1"
        >
          Item 2
        </span>
      </div>
    </FandomContentWell>
  </div>
</SimpleLocalNavigation>
`;
